src/NuGet.Protocol.Core.v3/DataClient/DataClient.cs

```diff
@@ -62,24 +70,33 @@ internal DataClient(HttpClientHandler handler, IEnumerable<INuGetMessageHandlerP
         /// <summary>
         /// Default caching handler used by the data client
         /// </summary>
-        public static HttpClientHandler DefaultHandler
+        public static HttpHandlerResourceV3 DefaultHandler
         {
-            get { return AssembleHandlers(CachingHandler, Enumerable.Empty<INuGetMessageHandlerProvider>()); }
+            get
+            {
+                // Client handler at the end of the pipeline
+                var clientHandler = CachingHandler;
+
+                return CreateHandler(clientHandler);
+            }
         }
 
         /// <summary>
-        /// Chain the handlers together
+        /// Default caching handler used by the data client
         /// </summary>
-        private static HttpClientHandler AssembleHandlers(HttpClientHandler handler, IEnumerable<INuGetMessageHandlerProvider> modifiers)
+        public static HttpHandlerResourceV3 CreateHandler(HttpClientHandler clientHandler)
         {
-            // final retry logic
-            // RetryHandler retryHandler = new RetryHandler(handler, 5);
-
-            // auth & proxy
-            //RequestModifierHandler modHandler = new RequestModifierHandler(handler, modifiers);
+            // Retry handler wrapping the client
+            var messageHandler = new RetryHandler(clientHandler, 5);
 
-            //return modHandler;
+            return new HttpHandlerResourceV3(clientHandler, messageHandler);
+        }
 
+        /// <summary>
+        /// Chain the handlers together
+        /// </summary>
+        private static HttpClientHandler AssembleHandlers(HttpClientHandler handler, IEnumerable<INuGetMessageHandlerProvider> modifiers)
```

Comment:
Looks like this private static method simply returns the HttpClientHandler 'handler'. Why does it take in an IEnumerable<INuGetMessageHandlerProvider>?